modules/augment/src/rotate.cpp

```diff
@@ -0,0 +1,40 @@
+// This file is part of OpenCV project.
+// It is subject to the license terms in the LICENSE file found in the top-level directory
+// of this distribution and at http://opencv.org/license.html.
+
+#include "precomp.hpp"
+namespace cv {
+namespace augment {
+
+    Rotate::Rotate(Range _angleRange)
+    {
+        this->angleRange = _angleRange;
+        this->resetRandom();
+    }
+
+    Rotate::Rotate() : Rotate::Rotate(Range(0, 180)) {}
+
+    void Rotate::resetRandom()
+    {
+        this->currentAngle = rng.uniform(this->angleRange.start, this->angleRange.end + 1);
+    }
+
+    void Rotate::image(InputArray _src, OutputArray _dst)
+    {
+        Mat src = _src.getMat();
+        Mat rotationMat = getRotationMatrix2D(Point2f(src.cols / 2, src.rows / 2), this->currentAngle, 1);
```

Comment:
I'd like to rename `resetRandom` to something like `init` because or now rotation matrix is computed everywhere. So please move it's initialization to `init`. All the random values should be initialized there as well.

Reply:
I renamed it to init.. now all transformations can call t.init(image)
then any transformation can be apply on any ground truth data without needing to send the image again with the data for example : t.point((x,y))

Reply:
and the rotation matrix is computed only once